src/message.rs

```diff
@@ -1433,3 +1433,17 @@ pub enum MessageType {
+        assert!(serde_json::from_str::<Message>(serialized_message).is_ok());
+        // extra test with `unlock_blocks` because of a breaking change
+        let serialized_message = r#"{"id":"263a908e9af5f76052d01e1196c645a0b3b68326c0cf6c83ceda9f284f81eb3f","version":1,"parents":["0e6bafd31ddd6ea5ab9ef86e3b93497a0af069ed979c3e46b44b7ed80f191a4f","27020fdf43e72693cfa4e2d5f29efac6e4f7b2f2d7e3442348e7a904f695372c","60e495f804e622d7342753d2017c85c15d98d869bf8c809404c17c6178e8047d","e1e9bac26cf0ede7092e7bdebf7274c0eb5cedb642ea24df13f06b2e3af121dd"],"payloadLength":233,"payload":{"type":"Transaction","data":{"essence":{"type":"Regular","data":{"inputs":[{"type":"Utxo","data":{"input":"1ac3a0d8d2aa4b945bc18e5439d7a064bd15c45c6436ef3ed3e0933018a1b6ca0000","metadata":{"transactionId":"1ac3a0d8d2aa4b945bc18e5439d7a064bd15c45c6436ef3ed3e0933018a1b6ca","messageId":"72c54ded382cabf4c96fe39be7ed8712c090d019b813a1edb24e0e05fc2cd796","index":0,"amount":100000000,"isSpent":false,"address":"atoi1qql3gq76fg66v5w4ymrfgmncxg9q53w2tamzrxe5423gpcvxml2ak2w6w52","kind":"SignatureLockedSingle"}}}],"outputs":[{"type":"SignatureLockedSingle","data":{"address":"atoi1qzjvshnp0ef2kxqnrm9gn0wh46sp5avmpmpf5cdv9wwj4qjxjc99zu92z8d","amount":90000000,"remainder":true}},{"type":"SignatureLockedDustAllowance","data":{"address":"atoi1qzt0nhsf38nh6rs4p6zs5knqp6psgha9wsv74uajqgjmwc75ugupx3y7x0r","amount":10000000}}],"payload":null,"internal":false,"incoming":false,"value":10000000,"remainderValue":90000000}},"unlock_blocks":[{"type":"Signature","data":{"type":"Ed25519","data":{"public_key":[82,103,140,222,67,242,235,58,45,172,133,44,251,146,12,105,89,164,74,214,105,11,159,248,250,182,55,27,117,246,222,101],"signature":[191,164,179,252,155,49,105,77,206,234,68,48,33,231,122,138,15,136,231,31,64,112,164,160,78,161,39,38,15,222,251,232,176,114,213,14,41,4,95,28,115,231,212,238,73,136,232,124,105,42,212,192,170,151,158,206,182,237,216,114,144,114,123,9]}}}]}},"timestamp":"2022-04-05T12:34:29.024451798Z","nonce":5764607523034346247,"broadcasted":true,"reattachmentMessageId":null}"#;
+        assert!(serde_json::from_str::<Message>(serialized_message).is_ok());
```

Comment:
I would also check that they are actually equal, just in case there is some default action if the key is not found or whatever